infra/terraform/modules/tezos_node/security_group.tf

```diff
@@ -41,7 +41,8 @@ resource "aws_security_group_rule" "rpc_ingress_for_tezos_node" {
   security_group_id = aws_security_group.tezos_node.id
 }
 
-resource "aws_security_group_rule" "p2p_ingress_for_tezos_node" {
+resource "aws_security_group_rule" "p2p_ingress_for_tezos_mainnet_node" {
+  count       = var.TZ_NETWORK == "carthagenet" ? 0 : 1
```

Comment:
variable "count" wrongly named ?

Reply:
With terraform, there is no "if" condition. If you want to generate a resource or not, you have to use the generic keyword `count` which permits to define if you want 0, 1 or more resources.
If I want to translate this line, it could be something like this : "if TZ_NETWORK is equal to carthagenet then create 0 resource with this definition, else generate 1 with this defintion".

https://www.terraform.io/docs/configuration/resources.html#the-count-object